### PR TITLE
RoundRobinInterceptor implementation

### PR DESCRIPTION
**Pull request by justinbchan**
Implement round robin interceptor; add tests.

Note: per https://github.com/palantir/http-remoting/pull/729, QoS unavailable exceptions will not mark a URL as failed, so some original QoS tests are not expected to pass in the case of `NodeSelectionStrategy.ROUND_ROBIN`; as such, tests of QoS nature only use `NodeSelectionStrategy.PIN_UNTIL_ERROR` cases.

### CODE REVIEW

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UrlSelector.java [hunk 1]
```diff
@@ -45,6 +45,12 @@
      */
     Optional<HttpUrl> redirectToCurrent(HttpUrl current);
 
+    /**
+     * Similar to {@link #redirectTo}, but redirects the given URL to the next (in some undefined order) after the
+     * current.
+     */
+    Optional<HttpUrl> redirectToNextRoundRobin(HttpUrl current);
```
Comment (uschi2000) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UrlSelector.java]: not clear from the docs how this is different from redirectToNext?

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java [hunk 2]
```diff
@@ -130,8 +131,13 @@ private static RemotingOkHttpClient createInternal(
             // TODO(rfink): Should this go into the call itself?
             client.addInterceptor(new MeshProxyInterceptor(config.meshProxy().get()));
         } else {
-            // Add CurrentUrlInterceptor: always selects the "current" URL, rather than the one specified in the request
-            client.addInterceptor(CurrentUrlInterceptor.create(urlSelector));
+            if (config.nodeSelectionStrategy().equals(NodeSelectionStrategy.ROUND_ROBIN)) {
```
Comment (uschi2000) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java]: if you make this a `switch` then error-prone will validate that we check all enum cases

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RoundRobinUrlInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+package com.palantir.remoting3.okhttp;
+
+import java.io.IOException;
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+final class RoundRobinUrlInterceptor implements Interceptor {
```
Comment (uschi2000) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RoundRobinUrlInterceptor.java]: 2 sentence docs please

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java [hunk 4]
```diff
@@ -130,8 +130,16 @@ private static RemotingOkHttpClient createInternal(
...
             client.addInterceptor(new MeshProxyInterceptor(config.meshProxy().get()));
         } else {
-            // Add CurrentUrlInterceptor: always selects the "current" URL, rather than the one specified in the request
-            client.addInterceptor(CurrentUrlInterceptor.create(urlSelector));
+            switch (config.nodeSelectionStrategy()) {
+                case ROUND_ROBIN:
+                    client.addInterceptor(RoundRobinUrlInterceptor.create(urlSelector));
+                    break;
+                case PIN_UNTIL_ERROR:
+                default:
```
Comment (uschi2000) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java]: let's remove the `default`. error-prone will flag if this switch is incomplete. (can you verify, please?)

File: okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UrlSelectorImpl.java [hunk 5]
```diff
@@ -151,6 +145,16 @@ private static String switchWsToHttp(String url) {
         return redirectTo(current, baseUrls.get(currentUrl.get()));
     }
 
+    @Override
+    public Optional<HttpUrl> redirectToNextRoundRobin(HttpUrl current) {
```
Comment (uschi2000) [inline okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UrlSelectorImpl.java:149]: please 1-2 tests in UrlSelectorTest

**Review by uschi2000 — COMMENTED** [review]
two last comments, then I'm happy. (sorry for the delay.)

**Comment by justinbchan** [comment]
No worries -- pushed updates to address! Thanks @uschi2000 !

**Review by uschi2000 — APPROVED** [review]

**Pull request merged**